ksql-serde/src/main/java/io/confluent/ksql/serde/avro/KsqlGenericRowAvroSerializer.java

```diff
@@ -75,11 +76,14 @@ public void configure(final Map<String, ?> map, final boolean b) {
     try {
       GenericRecord avroRecord = new GenericData.Record(avroSchema);
       for (int i = 0; i < genericRow.getColumns().size(); i++) {
-        if (fields.get(i).schema().getType() == Schema.Type.ARRAY) {
-          avroRecord.put(
-              fields.get(i).name(),
-              Arrays.asList((Object[]) genericRow.getColumns().get(i))
-          );
+        Schema schema = getNonNullSchema(fields.get(i).schema());
```

Comment:
It seems like `getNonNullSchema` is just used to extract the inner schema and check if it is an array. Would it make more sense to rename the method `isNonNullArrayField` and return a boolean? This would push the details into the helper method and would make the code easier to read IMO.